internal/telemetry/componentattribute/instrument_int.go

```
@@ -0,0 +1,135 @@
+	return int64Histogram{
+		Int64Histogram: inst,
+		option:         mwa.option,
+	}, err
+}
+
+func (inst int64Histogram) Record(ctx context.Context, incr int64, options ...metric.RecordOption) {
+	options = slices.Insert(options, 0, metric.RecordOption(inst.option))
+	inst.Int64Histogram.Record(ctx, incr, options...)
+}
+
+type int64Gauge struct {
+	metric.Int64Gauge
+	option metric.MeasurementOption
+}
+
+func (mwa meterWithAttributes) Int64Gauge(name string, options ...metric.Int64GaugeOption) (metric.Int64Gauge, error) {
+	inst, err := mwa.Meter.Int64Gauge(name, options...)
+	return int64Gauge{
+		Int64Gauge: inst,
+		option:     mwa.option,
+	}, err
+}
+
+func (inst int64Gauge) Record(ctx context.Context, incr int64, options ...metric.RecordOption) {
+	options = slices.Insert(options, 0, metric.RecordOption(inst.option))
+	inst.Int64Gauge.Record(ctx, incr, options...)
+}
+
+type int64Observer struct {
+	metric.Int64Observer
+	option metric.MeasurementOption
+}
+
+func (obs int64Observer) Observe(value int64, options ...metric.ObserveOption) {
+	options = slices.Insert(options, 0, metric.ObserveOption(obs.option))
+	obs.Int64Observer.Observe(value, options...)
+}
+
+func wrapInt64Callback(cb metric.Int64Callback, option metric.MeasurementOption) metric.Int64Callback {
+	return func(ctx context.Context, io metric.Int64Observer) error {
+		return cb(ctx, int64Observer{Int64Observer: io, option: option})
+	}
+}
+
+func (mwa meterWithAttributes) Int64ObservableCounter(name string, options ...metric.Int64ObservableCounterOption) (metric.Int64ObservableCounter, error) {
+	cfg := metric.NewInt64ObservableCounterConfig(options...)
+	newOptions := []metric.Int64ObservableCounterOption{
+		metric.WithUnit(cfg.Unit()),
+		metric.WithDescription(cfg.Description()),
+	}
+	for _, cb := range cfg.Callbacks() {
+		newOptions = append(newOptions, metric.WithInt64Callback(wrapInt64Callback(cb, mwa.option)))
+	}
+	return mwa.Meter.Int64ObservableCounter(name, newOptions...)
+}
```

This method of wrapping the callbacks (using `NewXConfig` to parse the options, wrapping the callbacks, then manually "reserializing" the config) will be somewhat problematic if the API adds a new kind of option for asynchronous instruments, as we will be silently dropping them until we update this code. Does anyone have a better way to do this?